File: app/src/main/kotlin/template/ui/reader/viewer/pager/Pager.java
package template.ui.reader.viewer.pager;

import androidx.viewpager.widget.PagerAdapter;
import android.view.ViewGroup;

import io.reactivex.functions.Consumer;

public interface Pager {

    void setId(int id);
    void setLayoutParams(ViewGroup.LayoutParams layoutParams);

    void setOffscreenPageLimit(int limit);

    int getCurrentItem();
    void setCurrentItem(int item, boolean smoothScroll);

    int getWidth();
    int getHeight();

    abstract PagerAdapter getAdapter();
    void setAdapter(PagerAdapter adapter);

    void setOnChapterBoundariesOutListener(OnChapterBoundariesOutListener listener);

    void setOnPageChangeListener(Consumer<Integer> onPageChanged); // kotlin : fun setOnPageChangeListener(onPageChanged: (Int) -> Unit)
    void clearOnPageChangeListeners();
}

File: app/src/main/kotlin/template/ui/reader/viewer/pager/Pager.kt
package template.ui.reader.viewer.pager

import android.support.v4.view.PagerAdapter
import android.view.ViewGroup
import io.reactivex.functions.Consumer

/**
 * Created by Robin Yeung on 8/25/18.
 */
public interface Pager {

    fun setId(id: Int)
    fun setLayoutParams(layoutParams: ViewGroup.LayoutParams)

    fun setOffscreenPageLimit(limit: Int)

    fun getCurrentItem(): Int
    fun setCurrentItem(item: Int, smoothScroll: Boolean)

    fun getWidth(): Int
    fun getHeight(): Int

    fun getAdapter(): PagerAdapter
    fun setAdapter(adapter: PagerAdapter)

    fun setOnChapterBoundariesOutListener(listener: OnChapterBoundariesOutListener)

    fun setOnPageChangeListener(onPageChanged: Consumer<Int>)
    fun clearOnPageChangeListeners()
}